src/components/Subscribepost/Subscribepost.jsx

```diff
@@ -0,0 +1,224 @@
+
+        const newData = data.map((item) => {
+          if (item._id === result._id) {
+            return result;
+          } else {
+            return item;
+          }
+        });
+
+        setData(newData);
+      })
+      .catch((err) => console.log(err));
+  };
+
+  const deletePost = (postId) => {
+    fetch(`https://chit-chat-12.herokuapp.com/delete/${postId}`, {
+      method: "delete",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: "Bearer " + localStorage.getItem("jwt"),
+      },
+    })
+      .then((res) => res.json())
+      .then((result) => {
+        console.log(result);
+
+        const newData = data.filter((elem) => {
+          return elem._id !== result._id;
+        });
+
+        setData(newData);
+      })
+      .catch((err) => console.log(err));
+  };
+
+  
+  return (
+    <div className="body">
+      <div className="media_content">
+        {/* Card */}
+
+        {data &&
+          data.map((elem) => {
+            return (
+              <div className="media-card">
+                <div>
+                  <div
+                    style={{
+                      backgroundColor: "white",
+                      display: "flex",
+                      justifyContent: "space-between",
+                    }}
+                  >
+                    <h5
+                      style={{ paddingLeft: "0.5rem", paddingBottom: "1rem" }}
+                    >
+                      {" "}
+                      <Link
+                        to={
+                          elem.postedBy._id !== state._id
+                            ? "/profile/" + elem.postedBy._id
+                            : "/profile"
+                        }
+                      >
+                        {elem.postedBy.name}
+                      </Link>{" "}
+                    </h5>
+
+                    <DeleteIcon />
+                  </div>
+                  <img src={elem.photo} alt="" />
+                  <h3 style={{ paddingLeft: "0.5rem" }}>{elem.title}</h3>
+
+                  {elem.likes.includes(state._id) ? (
+                    <IconButton>
+                      <ThumbDownAltIcon onClick={() => postUnlikes(elem._id)} />
+                    </IconButton>
+                  ) : (
+                    <IconButton>
+                      <ThumbUpIcon onClick={() => postLikes(elem._id)} />
+                    </IconButton>
+                  )}
+
+                  <h5 style={{ paddingLeft: "0.5rem" }}>
+                    {elem.likes.length} likes
+                  </h5>
+
+                  <h5 style={{ paddingLeft: "0.5rem" }}>{elem.body}</h5>
+
+                  {elem.comments.map((item) => {
+                    return (
+                      <h5>
+                        {" "}
+                        <span
+                          style={{ fontWeight: "400", paddingLeft: "0.5rem" }}
+                        >
+                          {item.postedBy.name}-
```

Comment:
This route is not getting rendered.
It raises an error.
![image](https://user-images.githubusercontent.com/56365512/124745428-7098e700-df3d-11eb-821b-39bde4d78045.png)